### PR TITLE
spec/reactors/mempool: batch txs per peer

### PR DESCRIPTION
**Pull request by melekes**
Refs https://github.com/tendermint/tendermint/issues/625

### CODE REVIEW

**Review by marbar3778 — APPROVED** [review]
LGTM. Do we want one approval from IG and one from Informal?

File: spec/reactors/mempool/reactor.md [hunk 1]
```diff
@@ -13,6 +13,11 @@ for details.
 Sending incorrectly encoded data or data exceeding `maxMsgSize` will result
 in stopping the peer.
 
+`maxMsgSize` equals `MaxTxBytes` (1MB) * `TxBatchSize` (100) + 4 (proto
```
Comment (ValarDragon) [inline spec/reactors/mempool/reactor.md:16]: Can we set maxMsgSize as a config parameter independently from MaxTxBytes and TxBatchSize?

e.g. suppose I expect all txs to be 1KB, want a batch size of 1000, and want to allow occasional txs at the maxTxBytes (1MB).

In this setting, it seems reasonable to me to habe a maxMsgSize of around 1 to 2 MB, whereas this parameterization would net me 1GB.

File: spec/reactors/mempool/reactor.md [hunk 2]
```diff
@@ -13,6 +13,11 @@ for details.
...
+overhead). `MaxTxBytes` and `TxBatchSize` are mempool config parameters ->
+defined locally. The reactor sends transactions to the connected peers in
+batches. The maximum size of one batch is `TxBatchSize`.
```
Comment (ValarDragon) [inline spec/reactors/mempool/reactor.md:19]: Do peers get told your MaxTxBytes and TxBatchSize in the handshake?

**Review by ValarDragon — CHANGES_REQUESTED** [review]
Glad that batch mempool txs are getting added!

**Comment by melekes** [comment]
> LGTM. Do we want one approval from IG and one from Informal?

sure

File: spec/reactors/mempool/reactor.md (hunk 1 above)
Comment (melekes) [reply spec/reactors/mempool/reactor.md]: sorry, I just updated the spec to align with latest implementation. Now, there's a single parameter for maxMsgSize - MaxBatchSize (10MB).

File: spec/reactors/mempool/reactor.md (hunk 2 above)
Comment (melekes) [reply spec/reactors/mempool/reactor.md]: Not yet -> https://github.com/tendermint/tendermint/issues/5332

**Review by alexanderbez — APPROVED** [review]

**Review by brapse — APPROVED** [review]
👍

**Pull request merged**